core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -404,8 +404,11 @@ public int getLostWorkerCount() {
   public long getCapacityBytes() {
     long ret = 0;
     for (MasterWorkerInfo worker : mWorkers) {
-      synchronized (worker) {
+      worker.getUsageLock().readLock().lock();
```

Comment:
how about following the code convention in `DefautlFileSystemMaster` and returning a `LockResrouce` instead of the lock, so the code is more error-prune 
( I don't have to check the `finally` block and the order)

```
try (LockResource r = new LockResource(worker.getUsageLock().readLock())) {
  ret += worker.getCapacityBytes();
}
```

Reply:
Oh I didn't know that. Thanks will do!

Reply:
@apc999  I've applied `LockResource` whenever I only have a single lock. For multiple locks the order does matter I think. The order or acquiring multiple locks should be uniformed across functions and the order or releasing should be the opposite, in order to guard deadlocks. How do I use the `LockResource` to achieve that?